### ISSUE TITLE
[MCHECKSTYLE-405] The checkstyle plugin should report the number of all violations

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-checkstyle-plugin#317]
**[Guillaume Nodet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gnt)** opened **[MCHECKSTYLE-405](https://issues.apache.org/jira/browse/MCHECKSTYLE-405?redirect=false)** and commented

When there is an error, the build looks like:

```java
[INFO] There is 1 error reported by Checkstyle 8.41.1 with config/maven_checks.xml ruleset.
[ERROR] src/main/java/org/apache/maven/execution/BuildResumptionDataRepository.java:[54,30] (whitespace) ParenPad: '(' is not followed by whitespace.
 
```

and when this error is fixed, the build then looks like:

```
[INFO] You have 0 Checkstyle violations.
```

The problem is that checkstyle never caches the results for files with violations, even at info or warning level.  This cause the files to be always processed, without the user really being aware of any problem.

The number of violations at info or warning level should also be printed.  In addition, if there are errors, the output should be logged at WARN level.

---

1 votes, 2 watchers

**Comment by jira-importer** [comment]
**[Guillaume Nodet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gnt)** commented

The main problem is actually caused by the `violationSeverity` configuration parameter which silently ignores some violations.  Any file containing a violation can not be cached by checkstyle, so the result is that the user does not see any violation, but files are not cached.

See https://github.com/apache/maven/pull/477/commits/ec1b78b59d5e23b050cd1cf029efb9cf72917ac9

**Comment by jira-importer** [comment]
**[Jeff Thomas](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jwt007)** commented

Came here looking for this exact issue :)

I also find the reported number of violations '0' when "`{}failOnViolation=false{`}" misleading. 

I am a wee bit OCD so I want the count even if I don't want to break the build on violations. :P
